### PR TITLE
MULE-11387: Add Flow name as part of the HTTP policy pointcut parameters

### PR DESCRIPTION
**Pull request by balbifm**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java [hunk 1]
```diff
@@ -18,19 +18,23 @@
...
 
   /**
    * Creates a new {@link PolicyPointcutParameters}
-   *
+   * 
    * @param componentIdentifier the component identifier. This is the namespace of the module were it is defined and the source /
    *        operation identifier.
    * @param path the target path of the http:request operation.
    * @param method the HTTP method of the http:request operation.
+   * @param flowName name of the flow executing the http request
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java]: http > HTTP

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java [hunk 2]
```diff
@@ -46,4 +50,11 @@ public String getPath() {
   public String getMethod() {
     return method;
   }
+
+  /**
+   * @return the name of the flow executing the http request
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java]: http > HTTP

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java [hunk 3]
```diff
@@ -46,4 +50,11 @@ public String getPath() {
...
+   */
+  public String getFlowName() {
+    return flowName;
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java]: What is the flow name used for? Doe sit make sense to have it in PolicyPointcutParameters instead?

Comment (balbifm) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java]: We are using the flow name in our AutoDiscovery element to map the API in Platform with the implementation. So we need the flow name to compare against the one in the AutoDiscovery element when a policy has an API Pointcut. 
We have in mind to re-think that element, so not sure if we are going to need it after that, but we need it for the next release at least

File: core/src/main/java/org/mule/runtime/core/policy/OperationPolicyPointcutParametersFactory.java [hunk 4]
```diff
@@ -30,9 +30,10 @@
    *
    * @param operationIdentifier identifier of the message source
    * @param operationParameters set of parameters that are going to be used to execute the operation.
+   * @param flowName name of the flow executing the operation.
    * @return the pointcut parameters.
    */
   PolicyPointcutParameters createPolicyPointcutParameters(ComponentIdentifier operationIdentifier,
-                                                          Map<String, Object> operationParameters);
+                                                          Map<String, Object> operationParameters, String flowName);
```
Comment (pablokraan) [inline core/src/main/java/org/mule/runtime/core/policy/OperationPolicyPointcutParametersFactory.java]: Chang the signature to make flowName the first parameter

File: core/src/main/java/org/mule/runtime/core/policy/PolicyPointcutParameters.java [hunk 5]
```diff
@@ -19,15 +19,18 @@
...
 
   /**
    * Creates a new {@link PolicyPointcutParameters}
    * 
    * @param componentIdentifier the component identifier. This is the namespace of the module were it is defined and the source /
    *        operation identifier.
+   * @param flowName name of the flow where the source / operation is defined
    */
-  public PolicyPointcutParameters(ComponentIdentifier componentIdentifier) {
+  public PolicyPointcutParameters(ComponentIdentifier componentIdentifier, String flowName) {
```
Comment (pablokraan) [inline core/src/main/java/org/mule/runtime/core/policy/PolicyPointcutParameters.java]: Same here

File: core/src/main/java/org/mule/runtime/core/policy/OperationPolicyPointcutParametersFactory.java (hunk 4 above)
Comment (pablokraan) [inline core/src/main/java/org/mule/runtime/core/policy/OperationPolicyPointcutParametersFactory.java]: Looks like flowName should be non empty (ie: javadoc must be updated and validated properly validated)

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyPointcutParameters.java [hunk 6]
```diff
@@ -21,14 +21,16 @@
...
+   * 
    * @param componentIdentifier the component identifier. This is the namespace of the module were it is defined and the source /
    *        operation identifier.
+   * @param flowName name of the flow where the listener is defined
    * @param path the target path of the incoming request
    * @param method the HTTP method of the incoming request
    */
-  public HttpListenerPolicyPointcutParameters(ComponentIdentifier componentIdentifier, String path, String method) {
-    super(componentIdentifier);
+  public HttpListenerPolicyPointcutParameters(ComponentIdentifier componentIdentifier, String flowName, String path,
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyPointcutParameters.java]: Same comments from previous flowName parameters apply here

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyPointcutParametersFactory.java [hunk 7]
```diff
@@ -29,12 +29,13 @@ public boolean supportsSourceIdentifier(ComponentIdentifier sourceIdentifier) {
   }
 
   @Override
-  public PolicyPointcutParameters createPolicyPointcutParameters(ComponentIdentifier sourceIdentifier, Attributes attributes) {
+  public PolicyPointcutParameters createPolicyPointcutParameters(ComponentIdentifier sourceIdentifier, Attributes attributes,
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyPointcutParametersFactory.java]: Same here

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java [hunk 8]
```diff
@@ -21,14 +21,16 @@
...
+   * 
    * @param componentIdentifier the component identifier. This is the namespace of the module were it is defined and the source /
    *        operation identifier.
+   * @param flowName name of the flow where the requester is defined
    * @param path the target path of the http:request operation.
    * @param method the HTTP method of the http:request operation.
    */
-  public HttpRequestPolicyPointcutParameters(ComponentIdentifier componentIdentifier, String path, String method) {
-    super(componentIdentifier);
+  public HttpRequestPolicyPointcutParameters(ComponentIdentifier componentIdentifier, String flowName, String path,
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParameters.java]: And here

File: modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParametersFactory.java [hunk 9]
```diff
@@ -31,9 +31,10 @@ public boolean supportsOperationIdentifier(ComponentIdentifier operationIdentifi
 
   @Override
   public PolicyPointcutParameters createPolicyPointcutParameters(ComponentIdentifier operationIdentifier,
-                                                                 Map<String, Object> operationParameters) {
+                                                                 Map<String, Object> operationParameters,
+                                                                 String flowName) {
```
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/extension/http/api/policy/HttpRequestPolicyPointcutParametersFactory.java]: Same here too

**Pull request merged**